zerver/tests/test_scim.py

```diff
@@ -830,15 +833,465 @@ class TestSCIMGroup(SCIMTestCase):
+        }
+        result = self.client_post(
+            "/scim/v2/Groups", payload, content_type="application/json", **self.scim_headers()
+        )
+        self.assertEqual(result.status_code, 201)
+        output_data = orjson.loads(result.content)
+
+        self.assertEqual(NamedUserGroup.objects.count(), original_group_count + 2)
+
+        new_group = NamedUserGroup.objects.latest("id")
+        self.assertEqual(new_group.name, "New group with members")
+        self.assertEqual(set(get_user_group_direct_member_ids(new_group)), {hamlet.id})
+
+        expected_response_schema = self.generate_group_schema(new_group)
+        self.assertEqual(expected_response_schema, output_data)
+
+    def test_put_change_name_and_members(self) -> None:
+        realm = get_realm("zulip")
+        hamlet = self.example_user("hamlet")
+        desdemona = self.example_user("desdemona")
+        othello = self.example_user("othello")
+
+        test_group = check_add_user_group(
+            realm, "Test group", [hamlet, desdemona], acting_user=desdemona
+        )
+
+        # PUT replaces all specified attributes of the group. Thus,
+        # this payload will replace both the name and set of members of
+        # the group.
+        payload = {
+            "schemas": ["urn:ietf:params:scim:schemas:core:2.0:Group"],
+            "id": test_group.id,
+            "displayName": "New test group name",
+            "members": [{"value": str(hamlet.id)}, {"value": str(othello.id)}],
+        }
+        result = self.json_put(f"/scim/v2/Groups/{test_group.id}", payload, **self.scim_headers())
+        self.assertEqual(result.status_code, 200)
+
+        test_group.refresh_from_db()
+        self.assertEqual(test_group.name, "New test group name")
+        new_member_ids = get_user_group_direct_member_ids(test_group)
+        self.assertEqual(set(new_member_ids), {hamlet.id, othello.id})
+
+        output_data = orjson.loads(result.content)
+        expected_response_schema = self.generate_group_schema(test_group)
+        self.assertEqual(output_data, expected_response_schema)
+
+    def test_patch_add_remove_operations(self) -> None:
+        realm = get_realm("zulip")
+        hamlet = self.example_user("hamlet")
+        desdemona = self.example_user("desdemona")
+        othello = self.example_user("othello")
+        cordelia = self.example_user("cordelia")
+        iago = self.example_user("iago")
+
+        test_group = check_add_user_group(
+            realm, "Test group", [hamlet, desdemona], acting_user=desdemona
+        )
+
+        payload = {
+            "schemas": ["urn:ietf:params:scim:api:messages:2.0:PatchOp"],
+            "Operations": [
+                {
+                    "op": "add",
+                    "path": "members",
+                    # Include a user who's already in the group in this list, to verify that trying to add
+                    # a user who's already in the group is gracefully ignored.
```

Comment:
It'd be nice if this comment were above just the one user for whom this applies.